packages/adblocker/src/lists.ts

```diff
@@ -189,18 +202,55 @@ export function parseFilters(
       const filter = NetworkFilter.parse(line, config.debug);
       if (filter !== null) {
         networkFilters.push(filter);
+        if (preprocessorStack.length) {
+          filterStack.push(filter);
+        }
       }
     } else if (filterType === FilterType.COSMETIC && config.loadCosmeticFilters === true) {
       const filter = CosmeticFilter.parse(line, config.debug);
       if (filter !== null) {
         if (config.loadGenericCosmeticsFilters === true || filter.isGenericHide() === false) {
           cosmeticFilters.push(filter);
+          if (preprocessorStack.length) {
+            filterStack.push(filter);
+          }
+        }
+      }
+    } else if (config.loadPreprocessors) {
+      const preprocessorType = detectPreprocessor(line);
+
+      if (preprocessorType === PreprocessorTypes.BEGIF) {
+        preprocessorStack.push({
+          start: filterStack.length,
+          condition: Preprocessor.getCondition(line),
+        });
+      } else if (
+        (preprocessorType === PreprocessorTypes.ELSE ||
+          preprocessorType === PreprocessorTypes.ENDIF) &&
+        preprocessorStack.length
+      ) {
+        const last = preprocessorStack.pop()!;
+        const filters: Set<number> = new Set();
+
+        const breakAt = last.start + 1;
+
+        while (filterStack.length === breakAt) {
+          filters.add(filterStack.pop()!.getId());
+        }
+
+        preprocessors.push(Preprocessor.parse(last.condition, filters));
+
+        if (preprocessorType === PreprocessorTypes.ELSE) {
+          preprocessorStack.push({
+            start: filterStack.length,
+            condition: `!(${last.condition})`,
+          });
```

Comment:
I think this happened by invalid testing code. I've updated testing code to avoid using constants such as `true` and `false`. Now, the test code should validate those conditions. However, all conditions will be merged and evaluated with logical AND in the updated implementation (which based on the above code snippet).